### ISSUE TITLE
Crash on snapshot deploy and serverstop/serverstart

### ISSUE THREAD

**Issue by lexesv** [TeaSpeak/TeaSpeak#55]
## System Information
***Operating System***: Debian 9
***Operating System Type*** (x86|x64): x64
***TeaSpeak Version***:  v1.0.29-beta

## Issue
***Crash on snapshot (from TS3 server) deploy***:
note: I use ts3admin.class
[ solved trim($snapshot); but crash ... ) ]

```
[13:41:53] [INFO]     | Having new server user on server 2. Database ID: 1 Name: serveradmin
[13:41:53] [ERROR]     | Could not apply server snapshot: Invalid tag in permissions. Available: end_permissions
[13:41:53] [ERROR] = 
---------------[ Crash log ]---------------
 A fatal signal got caught!
 Try to stop the application normaly
 Please send this crash report to the application maintainer WolverinDEV

  General Info:
     Version           : 1.0.29-beta
     Current Timestamp : 19-02-2018 01:41:53
     System version    : sn: Linux nn r: 4.9.0-5-amd64 v: #1 SMP Debian 4.9.65-3+deb9u2 (2018-01-04) m: x86_64 d: (none)

  Error Info:
    Error type        : Signal
    Number            : 0
    Message           : Success
    Code              : 1

  Backtrace:
[ 1]    [./TeaSpeakServer                     ] [0x000000000073cd0e] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[ 2]    [./TeaSpeakServer                     ] [0x00000000004d77ee] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 3]    [./TeaSpeakServer                     ] [0x00000000004d3fa6] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 4]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007ff5a27ac0c0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x110c0) [0x7ff5a27ac0c0]
[ 5]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007ff5a27a4b20] /lib/x86_64-linux-gnu/libpthread.so.0(pthread_mutex_lock+0) [0x7ff5a27a4b20]
[ 6]    [./TeaSpeakServer                     ] [0x0000000000444272] ./TeaSpeakServer() [0x444272]
[ 7]    [./TeaSpeakServer                     ] [0x000000000044897c] std::mutex::lock()                                                                                                                                                                                                                                                                                  at ??:0
[ 8]    [./TeaSpeakServer                     ] [0x000000000044debc] std::lock_guard<std::mutex>::lock_guard(std::mutex&)                                                                                                                                                                                                                                                at ??:0
[ 9]    [./TeaSpeakServer                     ] [0x00000000004b6641] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[10]    [./TeaSpeakServer                     ] [0x0000000000655088] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[11]    [./TeaSpeakServer                     ] [0x000000000064773c] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[12]    [./TeaSpeakServer                     ] [0x0000000000641e16] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[13]    [./TeaSpeakServer                     ] [0x0000000000641879] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[14]    [./TeaSpeakServer                     ] [0x000000000063f564] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[15]    [./TeaSpeakServer                     ] [0x000000000063f594] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[16]    [./TeaSpeakServer                     ] [0x00000000007b246c] std::experimental::filesystem::v1::path::_M_split_cmpts()                                                                                                                                                                                                                                           at ??:0
[17]    [./TeaSpeakServer                     ] [0x000000000062da42] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[18]    [./TeaSpeakServer                     ] [0x000000000063364e] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[19]    [./TeaSpeakServer                     ] [0x0000000000633582] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[20]    [./TeaSpeakServer                     ] [0x0000000000633512] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[21]    [./TeaSpeakServer                     ] [0x0000000000736667] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[22]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007ff5a27a2494] /lib/x86_64-linux-gnu/libpthread.so.0(+0x7494) [0x7ff5a27a2494]
[23]    [/lib/x86_64-linux-gnu/libc.so.6      ] [0x00007ff5a186daff] /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7ff5a186daff]

```

***crash on serverstart/serverstop***
after:
serveredit serveredit virtualserver_maxclients=30
error id=2 msg=not\simplemented
serverstop sid=1
error id=0 msg=ok
serverstart sid=1
error id=0 msg=ok
serverstop sid=1
error id=0 msg=ok
serverstart sid=1
error id=0 msg=ok
Connection closed by foreign host.

```
[17:16:56] [INFO]   1 | New client joined. (UID: OHc0Sdv8Hf9YX0bz7ygsIs5JFok= Name: User1 Ip: *********:26688)
[17:17:16] [ERROR]   1 | Found delay while ticking server. (Delay: 9149ms)
---------------[ Crash log ]---------------
 A fatal signal got caught!
 Try to stop the application normaly
 Please send this crash report to the application maintainer WolverinDEV

  General Info:
     Version           : 1.0.29-beta
     Current Timestamp : 19-02-2018 05:17:16
     System version    : sn: Linux nn:  4.9.0-5-amd64 v: #1 SMP Debian 4.9.65-3+deb9u2 (2018-01-04) m: x86_64 d: (none)

  Error Info:
    Error type        : Signal
    Number            : 0
    Message           : Success
    Code              : 1

  Backtrace:
[ 1]    [./TeaSpeakServer                     ] [0x000000000073cd0e] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[ 2]    [./TeaSpeakServer                     ] [0x00000000004d77ee] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 3]    [./TeaSpeakServer                     ] [0x00000000004d3fa6] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 4]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007f42af4240c0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x110c0) [0x7f42af4240c0]
[ 5]    [./TeaSpeakServer                     ] [0x00000000004dd9f2] std::_Setfill<char> std::setfill<char>(char)                                                                                                                                                                                                                                                        at ??:0
[ 6]    [./TeaSpeakServer                     ] [0x00000000004d98e4] std::_Setfill<char> std::setfill<char>(char)                                                                                                                                                                                                                                                        at ??:0
[ 7]    [./TeaSpeakServer                     ] [0x00000000004e3925] std::_Setfill<char> std::setfill<char>(char)                                                                                                                                                                                                                                                        at ??:0
[ 8]    [./TeaSpeakServer                     ] [0x00000000004e85d7] std::_Setfill<char> std::setfill<char>(char)                                                                                                                                                                                                                                                        at ??:0
[ 9]    [./TeaSpeakServer                     ] [0x000000000048a5de] void __gnu_cxx::new_allocator<threads::MutexData>::destroy<threads::MutexData>(threads::MutexData*)                                                                                                                                                                                                 at ??:0
[10]    [./TeaSpeakServer                     ] [0x00000000004cef27] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[11]    [./TeaSpeakServer                     ] [0x00000000006ca851] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[12]    [./TeaSpeakServer                     ] [0x00000000006ca0b9] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[13]    [./TeaSpeakServer                     ] [0x00000000006c95cb] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[14]    [./TeaSpeakServer                     ] [0x00000000006c828e] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[15]    [./TeaSpeakServer                     ] [0x00000000004a789a] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[16]    [./TeaSpeakServer                     ] [0x0000000000736fc7] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[17]    [./TeaSpeakServer                     ] [0x00000000007379d0] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[18]    [./TeaSpeakServer                     ] [0x00000000004a789a] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[19]    [./TeaSpeakServer                     ] [0x000000000072e89d] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[20]    [./TeaSpeakServer                     ] [0x0000000000735e97] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[21]    [./TeaSpeakServer                     ] [0x0000000000735e2b] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[22]    [./TeaSpeakServer                     ] [0x0000000000735d0a] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[23]    [./TeaSpeakServer                     ] [0x0000000000735c9a] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[24]    [./TeaSpeakServer                     ] [0x0000000000736667] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[25]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007f42af41a494] /lib/x86_64-linux-gnu/libpthread.so.0(+0x7494) [0x7f42af41a494]
[26]    [/lib/x86_64-linux-gnu/libc.so.6      ] [0x00007f42ae4e5aff] /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f42ae4e5aff]
---------------[ Crash log ]---------------
> 

```

**Comment by WolverinDEV** [comment]
~ Fixed first crash (happens when deploy fails, still tries to start a non existing server)


What did you for the second crash? Or was it just randomly?

**Comment by lexesv** [comment]
> What did you for the second crash? Or was it just randomly?

just stop/start the server.

I repeated the action,  and crash again.

serverstop sid=1
error id=0 msg=ok
serverstart sid=1
error id=0 msg=ok
Connection closed by foreign host.

```
[14:04:53] [ERROR]   1 | Found delay while ticking server. (Delay: 24362ms)
---------------[ Crash log ]---------------
 A fatal signal got caught!
 Try to stop the application normaly
 Please send this crash report to the application maintainer WolverinDEV

  General Info:
     Version           : 1.0.29-beta
     Current Timestamp : 20-02-2018 02:04:53
     System version    : sn: Linux nn: 4.9.0-5-amd64 v: #1 SMP Debian 4.9.65-3+deb9u2 (2018-01-04) m: x86_64 d: (none)

  Error Info:
    Error type        : Signal
    Number            : 0
    Message           : Success
    Code              : 1

  Backtrace:
[ 1]    [./TeaSpeakServer                     ] [0x000000000073cd0e] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[ 2]    [./TeaSpeakServer                     ] [0x00000000004d77ee] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 3]    [./TeaSpeakServer                     ] [0x00000000004d3fa6] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 4]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007fa3ca2c20c0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x110c0) [0x7fa3ca2c20c0]
[ 5]    [./TeaSpeakServer                     ] [0x00000000004dd9f2] std::_Setfill<char> std::setfill<char>(char)                                                                                                                                                                                                                                                        at ??:0
[ 6]    [./TeaSpeakServer                     ] [0x00000000004d98e4] std::_Setfill<char> std::setfill<char>(char)                                                                                                                                                                                                                                                        at ??:0
[ 7]    [./TeaSpeakServer                     ] [0x00000000004e3925] std::_Setfill<char> std::setfill<char>(char)                                                                                                                                                                                                                                                        at ??:0
[ 8]    [./TeaSpeakServer                     ] [0x00000000004e85d7] std::_Setfill<char> std::setfill<char>(char)                                                                                                                                                                                                                                                        at ??:0
[ 9]    [./TeaSpeakServer                     ] [0x000000000048a5de] void __gnu_cxx::new_allocator<threads::MutexData>::destroy<threads::MutexData>(threads::MutexData*)                                                                                                                                                                                                 at ??:0
[10]    [./TeaSpeakServer                     ] [0x00000000004cef27] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[11]    [./TeaSpeakServer                     ] [0x00000000006ca851] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[12]    [./TeaSpeakServer                     ] [0x00000000006ca0b9] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[13]    [./TeaSpeakServer                     ] [0x00000000006c95cb] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[14]    [./TeaSpeakServer                     ] [0x00000000006c828e] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[15]    [./TeaSpeakServer                     ] [0x00000000004a789a] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[16]    [./TeaSpeakServer                     ] [0x0000000000736fc7] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[17]    [./TeaSpeakServer                     ] [0x00000000007379d0] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[18]    [./TeaSpeakServer                     ] [0x00000000004a789a] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[19]    [./TeaSpeakServer                     ] [0x000000000072e89d] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[20]    [./TeaSpeakServer                     ] [0x0000000000735e97] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[21]    [./TeaSpeakServer                     ] [0x0000000000735e2b] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[22]    [./TeaSpeakServer                     ] [0x0000000000735d0a] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[23]    [./TeaSpeakServer                     ] [0x0000000000735c9a] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[24]    [./TeaSpeakServer                     ] [0x0000000000736667] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[25]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007fa3ca2b8494] /lib/x86_64-linux-gnu/libpthread.so.0(+0x7494) [0x7fa3ca2b8494]
[26]    [/lib/x86_64-linux-gnu/libc.so.6      ] [0x00007fa3c9383aff] /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7fa3c9383aff]
---------------[ Crash log ]---------------

```

**Comment by WolverinDEV** [comment]
Fixed second crash too :) (1.0.29-beta-2)

**Comment by WolverinDEV** [comment]
~ I recommend to use (1.0.29-beta-3) fixed another similar issue
